shap/plots/_beeswarm.py

```diff
@@ -157,8 +174,12 @@ def beeswarm(
     if feature_names is None:
         feature_names = np.array([labels["FEATURE"] % str(i) for i in range(num_features)])
 
+    fig = pl.gcf()
+    if ax is None:
+        ax = pl.gca()
```

Comment:
We should use the figure that belongs to `ax`, which in general may be different to the current figure.

```suggestion
    if ax is None:
        ax = pl.gca()
    fig = ax.get_figure()
    assert fig is not None  # type narrowing for mypy
```